homeassistant/components/stream/__init__.py

```diff
@@ -507,22 +521,28 @@ def _should_retry() -> bool:
     {
         vol.Optional(CONF_RTSP_TRANSPORT): vol.In(RTSP_TRANSPORTS),
         vol.Optional(CONF_USE_WALLCLOCK_AS_TIMESTAMPS): bool,
+        vol.Optional(CONF_EXTRA_PART_WAIT_TIME): cv.positive_float,
     }
 )
 
 
-def convert_stream_options(stream_options: dict[str, str | bool]) -> dict[str, str]:
+def convert_stream_options(
+    hass: HomeAssistant, stream_options: Mapping[str, str | bool | float]
+) -> tuple[dict[str, str], StreamSettings]:
     """Convert options from stream options into PyAV options."""
+    stream_settings = copy.copy(hass.data[DOMAIN][ATTR_SETTINGS])
```

Comment:
Would it work to put this in `create_stream` so that it starts with these defaults, then merges the caller supplied defaults? I don't think i have a rational explanation for why it feels right to me, but want your take.

Reply:
Do you just mean moving this line out from this function into `create_stream`?

Reply:
OK, i guess its just the generic config flow that was throwing me off.

Reply:
Yes, that usage should probably just be removed

Reply:
Changed it to put it inline and removed the call from generic